### ISSUE TITLE
additionalVolumeMounts removed 

### ISSUE THREAD

**Issue by windowsrefund** [k8s-at-home/library-charts#65]
The following used to be possible.
```
persistence:
  config:
    enabled: true
  media:
    enabled: true
    existingClaim: media

additionalVolumeMounts:
  - name: media
    mountPath: /downloads
  - name: media
    mountPath: /series
    subPath: series

```
Please restore additionalVolumeMounts or provide clear documentation on how to achieve this end state.

**Comment by carpenike** [comment]
Hi @windowsrefund,

Functionally still persists, but has moved within the `persistence` key. Changelog here -- https://github.com/k8s-at-home/library-charts/tree/main/charts/stable/common#3.0.0

Example configuration here: https://github.com/carpenike/k8s-gitops/blob/master/cluster/apps/home/zigbee2mqtt/helm-release.yaml#L45-L54

**Comment by windowsrefund** [comment]
It's not working though as I'm trying to mount my RWX pvc on both /series and /downloads. 

```
persistence:
  config:
    enabled: true
    skipuninstall: true
  series:
    enabled: true
    type: pvc
    existingClaim: media
    mountPath: /series
    subPath: series
  downloads:
    enabled: true
    type: pvc
    existingClaim: media
    mountPath: /downloads
    subPath: downloads
```
From the describe
`Warning  FailedMount  2m34s (x75 over 6h27m)  kubelet  Unable to attach or mount volumes: unmounted volumes=[downloads], unattached volumes=[config downloads series kube-api-access-wgjkq]: timed out waiting for the condition`

**Comment by bjw-s** [comment]
We will have to devise a way to integrate having multiple `subPath`/`mountPath` combo's with our current solution for persistence.

**Comment by windowsrefund** [comment]
Just to clarify why this is so important when working with a single RWX PVC that's used by nzbget/transmission and things like sonarr/radarr...

The downloader is going to mount the PVC as /downloads

*arr is going to mount as /media. This will be set to the root folder which is scanned for new media. Sonar then imports a series into /media/NameOfSeries. This is obviously not clean/tidy but it works. The real problem of this sloppy structure is seen from a Jellyfin/Emby/Plex perspective since you'd have to point it to /media (assuming access is intact). The application is going to recursively scan that directory resulting in "best guess" attempts to interpret the crazy file names under /media/completed (<- Where nzbget stores completed downloads). The result is a bunch of false positive anime episodes showing up in the application's dashboard.

So basically, the way around that would be to use subPath in order to have /media mounted against something like /media/series. That would work nicely but it creates a problem on the *arr side of things since now the application can't import because there's no visibility into the completed directory.

So basically, I don't see how anyone can possibly be using a RWX PVC shared among all relevant apps ever since additionalVolumeMounts was just cast aside.

additionalVolumeMounts was clean, and totally usable? What possible incentive could there have been to just dump it so casually with no testing or thought given to overall impact?

**Comment by carpenike** [comment]
So in your case, are you trying to use a single PVC across all 3 scenarios with different subPath definitions?

**Comment by windowsrefund** [comment]
Yes, the first comment shows what used to be possible before the recent change to this library.

**Comment by onedr0p** [comment]
I'll just point out how I am doing it... I know this isn't a _fix_ just an alternative way of doing what you are trying to do.

My folder structure on my NFS share is like this:

```
./Media
|__ Downloads
     |__ qbittorrent
|__ Library
     |__ Movies
     |__ Series
```

This `Media` folder is mounted into to Sonarr, Radarr, qBittorrent, sabnzbd, Plex etc all the same as `/media`. All apps see every folder.

**Comment by windowsrefund** [comment]
> > additionalVolumeMounts was clean, and totally usable. What possible incentive could there have been to just dump it so casually with no testing or thought given to overall impact?
> 
> Again, we didn't purposely "dump" this. There's a lot of changes and it's hard to test them all the way thru everyones different setups. We count on you and others to provide feedback, I am sure we will have a way for you to do this. Sit tight on the older versions of the charts until then... or...
> 
> I'll just point out how I am doing it... I know this isn't a _fix_ just an alternative way of doing what you are trying to do.
> 
> My folder structure on my NFS share is like this:
> 
> ```
> ./Media
> |__ Downloads
>      |__ qbittorrent
> |__ Library
>      |__ Movies
>      |__ Series
> ```
> 
> This `Media` folder is mounted into to Sonarr, Radarr, qBittorrent, sabnzbd, Plex etc all the same as `/media`. All those apps see every folder.
> 
> I only need 1 PVC and no `subPath`s for all apps that are doing work in these directories.

Your NFS implementation is quite different however, from that of a RWX PVC. I'm guessing this "style" also means you are then expected to configure each and every application to read and write from particular areas of the share you are mounting. Obviously, it works for you but it imposes a degree of micro-management and it also assumes people are working with a similar NFS solution. This is where the mistake was made in my opinion. Everything should be based around a RWX PVC rather than some kind of SAN solution. 

That said, I appreciate things are happening behind the scenes to restore the functionality.

All the best

**Comment by onedr0p** [comment]
> I'm guessing this "style" also means you are then expected to configure each and every application to read and write from particular areas of the share you are mounting. Obviously, it works for you but it imposes a degree of micro-management and it also assumes people are working with a similar NFS solution.

All those applications require user defined configuration of where to put or look for things, what are you talking about?

**Comment by windowsrefund** [comment]
For example, nzbget ships out of the box configured to download into /downloads/completed. If you're just using a single mountpoint of /media, you'd need to change that path to /media/downloads/completed. That's just one example. The point I'm making here is that you're clearly managing your NFS server exclusively... creating the directory structore beforhand and then tailing each application as needed rather than being able to benefit from a more dynamic approach.

**Comment by onedr0p** [comment]
> For example, nzbget ships out of the box configured to download into /downloads/completed. If you're just using a single mountpoint of /media, you'd need to change that path to /media/downloads/completed. That's just one example. The point I'm making here is that you're clearly managing your NFS server exclusively... creating the directory structore beforhand and then tailing each application as needed rather than being able to benefit from a more dynamic approach.

Do you use Windows or MacOS or a Linux and never change any default settings? The whole point of these apps are to configure them to your liking and use.

If you just want what comes out of box without any custom configuration on these applications you're doing it completely wrong IMO.

**Comment by onedr0p** [comment]
> Excuse me? How is making use of well established features resulting in a perfectly useful environment doing it wrong? Seriously, you guys just ripped up the codebase with no thought on impact and now you're telling me I'm wrong?
> 
> 

Nothing is wrong with you opening an issue on the library chart or your use of it. What I said was you not changing application settings is wrong. You're relying on default application settings that are meant to be changed to suit your environment.

**Comment by windowsrefund** [comment]
You're kinda just being silly at this point and a bit overly defensive. Nobody is saying I don't tweak my apps as needed. What I've shown a light on here is that you guys are clearly writing your charts assuming people are managing external NFS servers (in the way you do) rather than writing them to support native Kubernetes objects. I mean... these are Helm charts... you should be writing them to make the best use of PVCs, not external SANs that have to be micro-managed in a particular fashion. Maybe try and take something positive away from this instead of looking for a way to attack me just because I use Kubernetes correctly?

**Comment by onedr0p** [comment]
We have some ideas on how to fix it but it will take time, we will update this issue when it's fixed 👍🏼

**Comment by bjw-s** [comment]
I don't see anybody being attacked here. I see two different solutions to the same problem. Both with equal merit if you ask me. Using Kubernetes "correctly" can mean a gazilion different things to just as many users. 

As has been indicated several times by different people, there are some workaround you can apply. 

If you feel these are not valid for your use case you will be better off sticking with the old chart versions until we implement a solution.

**Comment by windowsrefund** [comment]
It's not working though as I'm trying to mount my RWX pvc on both /series and /downloads. 

```
persistence:
  config:
    enabled: true
    skipuninstall: true
  series:
    enabled: true
    type: pvc
    existingClaim: media
    mountPath: /series
    subPath: series
  downloads:
    enabled: true
    type: pvc
    existingClaim: media
    mountPath: /downloads
    subPath: downloads
```
From the describe
`Warning  FailedMount  2m34s (x75 over 6h27m)  kubelet  Unable to attach or mount volumes: unmounted volumes=[downloads], unattached volumes=[config downloads series kube-api-access-wgjkq]: timed out waiting for the condition`


> We have some ideas on how to fix it but it will take time, we will update this issue when it's fixed 👍🏼

Thank you
